src/v/cluster/rm_partition_frontend.cc

```diff
@@ -253,162 +253,17 @@ ss::future<begin_tx_reply> rm_partition_frontend::do_begin_tx(
-        })
-      .then(&rpc::get_ctx_data<prepare_tx_reply>)
-      .then([](result<prepare_tx_reply> r) {
-          if (r.has_error()) {
-              vlog(txlog.warn, "got error {} on remote prepare tx", r.error());
-              return prepare_tx_reply{tx_errc::timeout};
-          }
-
-          return r.value();
-      });
-}
-
-ss::future<prepare_tx_reply> rm_partition_frontend::prepare_tx_locally(
-  model::ntp ntp,
-  model::term_id etag,
-  model::partition_id tm,
-  model::producer_identity pid,
-  model::tx_seq tx_seq,
-  model::timeout_clock::duration timeout) {
-    vlog(
-      txlog.trace,
-      "processing name:prepare_tx, ntp:{}, etag:{}, pid:{}, tx_seq:{}, "
-      "coordinator:{}",
-      ntp,
-      etag,
-      pid,
-      tx_seq,
-      tm);
-    auto reply = co_await do_prepare_tx(ntp, etag, tm, pid, tx_seq, timeout);
-    vlog(
-      txlog.trace,
-      "sending name:prepare_tx, ntp:{}, etag:{}, pid:{}, tx_seq:{}, "
-      "coordinator:{}, ec:{}",
-      ntp,
-      etag,
-      pid,
-      tx_seq,
-      tm,
-      reply.ec);
-    co_return reply;
-}
-
-ss::future<prepare_tx_reply> rm_partition_frontend::do_prepare_tx(
-  model::ntp ntp,
-  model::term_id etag,
-  model::partition_id tm,
-  model::producer_identity pid,
-  model::tx_seq tx_seq,
-  model::timeout_clock::duration timeout) {
-    if (!is_leader_of(ntp)) {
-        return ss::make_ready_future<prepare_tx_reply>(
-          prepare_tx_reply{tx_errc::leader_not_found});
-    }
-
-    auto shard = _shard_table.local().shard_for(ntp);
-
-    if (!shard) {
-        return ss::make_ready_future<prepare_tx_reply>(
-          prepare_tx_reply{tx_errc::shard_not_found});
-    }
-
-    return _partition_manager.invoke_on(
-      *shard,
-      _ssg,
-      [ntp, etag, tm, pid, tx_seq, timeout](
-        cluster::partition_manager& mgr) mutable {
-          auto partition = mgr.get(ntp);
-          if (!partition) {
-              return ss::make_ready_future<prepare_tx_reply>(
-                prepare_tx_reply{tx_errc::partition_not_found});
-          }
-
-          auto stm = partition->rm_stm();
-
-          if (!stm) {
-              vlog(txlog.warn, "can't get tx stm of the {}' partition", ntp);
-              return ss::make_ready_future<prepare_tx_reply>(
-                prepare_tx_reply{tx_errc::stm_not_found});
-          }
-
-          return stm->prepare_tx(etag, tm, pid, tx_seq, timeout)
-            .then([](tx_errc ec) { return prepare_tx_reply{ec}; });
-      });
+  model::ntp,
+  model::term_id,
+  model::partition_id,
+  model::producer_identity,
+  model::tx_seq,
+  model::timeout_clock::duration) {
+    // prepare_tx is unsupported starting 23.3.x original transactions
+    // design used a preparing/prepare phase as an intent to commit
+    // transaction and it has been redesigned since then starting 22.3.0
+    // this RPC is no longer used.
+    return ss::make_exception_future<prepare_tx_reply>({});
```

Comment:
can we thrown meaningful exception here ?

Reply:
Done.